src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -640,17 +641,42 @@ protected DynamoModel(IStartConfiguration config)
             // If not, check the default filepath for the Python template (userDataFolder\PythonTemplate.py).
             // If that doesn't exist either or is empty, Dynamo falls back to the hard-coded Python script template.
             // We log the behaviour to make it easy for users to troubleshoot this.
-            if (string.IsNullOrEmpty(PreferenceSettings.PythonTemplateFilePath) || !File.Exists(PreferenceSettings.PythonTemplateFilePath))
+
+            // If a custom python template path doesn't already exists in the DynamoSettings.xml
+            if(string.IsNullOrEmpty(PreferenceSettings.PythonTemplateFilePath) || !File.Exists(PreferenceSettings.PythonTemplateFilePath))
             {
-                if (string.IsNullOrEmpty(pathManager.PythonTemplateFilePath) || !File.Exists(pathManager.PythonTemplateFilePath))
-                    Logger.Log(Resources.PythonTemplateInvalid);
-                else
+                try
+                {
+                    // To supply a custom python template host integrators should supply a 'DefaultStartConfiguration' config file
+                    // or create a new struct that inherits from 'DefaultStartConfiguration' making sure to set the 'PythonTemplatePath'
+                    // while passing the config to the 'DynamoModel' constructor.
+                    var configCast = (DefaultStartConfiguration)config;
```

Comment:
This is done because the `DynamoModel` constructor takes a `IStartConfiguration` config as its only param.  I believe modifying the interface (even adding a new property) would be an API breaking change, but feel free to correct me if I am wrong on this or there is a better strategy.

Reply:
I think you should do something like:
`defaultConfig = config as DefaultStartConfiguration` and then check for null because it's possible that the concreteType is NOT a DefaultStartConfiguration ... and that needs to be handled. (ie missing the python template field you just added)

Reply:
alternatively you can add a second interface and check if the object implements that interface, if it does - do the right thing and set the path.

Reply:
👍 Same options in https://jira.autodesk.com/browse/DYN-1644

Reply:
in general, I don't think casting like this is a great idea for the problem we are discussing as casting can throw - `as` and `is` should not.'

https://docs.microsoft.com/en-us/dotnet/api/system.invalidcastexception?view=netframework-4.7.2

Reply:
Thanks for the feedback guys.  I was not happy with this approach either but `as` cannot be used with a non-nullable value type which `DefaultStartConfiguration` is as a struct.  This is why I wrapped this logic in a try/catch returning the default template if an exception is thrown.  However, performing a check such as 
```C#
if(config is DefaultStartConfiguration)
{
    // Proceed in checking if a custom template path is available and valid
}
else
{
    // Use default template
}
```
is possible but does not guaranteed the PythonTemplatePath has been set.  Attempting to retrieve the `PythonTemplatePath` property will simply return null if it was not set and I verify this is not the case below before proceeding.  Long story short we can probably use `is` to verify the `config` is of the `DefaultStartConfiguration` type or of a type that inherits from it.  I don't think adding another struct is necessary just for this property. If the `config` is of another type we return the default template.  I believe the try/catch can also be removed with `is` as @mjkkirschner suggested since it will not throw.  What do you guys think about this approach? @QilongTang

Reply:
Sounds good to me.